Update Use Cases in Developer Guide (#37)
AY2425S1-CS2103T-F14a-2/tp: docs/DeveloperGuide.md

### Use cases

**Use case 1: View List of Customers**

* **System**: SellSavvy

> * **Actor**: USer

Reviewer: What is USer?

* **Use Case**: UC01 - View List of Customers

> 

Reviewer 2: Consider including guarantees? Based on course use case examples.

**MSS**

> 

Reviewer 2: As per course website use case topics, perhaps removing the whitespace after MSS and following subsections of use cases may be better?

1.  User chooses to view the list of customers.
2.  SellSavvy retrieves all customers from the database.
3.  SellSavvy displays all customers along with their details.

> 4.  Use case ends.

Reviewer 2: Consider removing list number from `use case ends` statement
```suggestion
3. Sellsavvy...
Use case ends.
```

**Use case 2: View List of Customers**

* **System**: SellSavvy

> * **Actor**: USer

Reviewer: What is USer?

* **Use Case**: UC02 - Add Customer

**MSS**

> 1.  User chooses to add a new customer.

Reviewer 2: ```suggestion
1.  User wants to add a new customer.
```

2.  SellSavvy validates the input and adds the customer to the database.
3.  SellSavvy confirms the addition by displaying the newly added customer's details.
4.  Use case ends.

**Extensions**

* 1a. SellSavvy detects required details missing.
  * 1a1. SellSavvy displays an error message “Missing details of customer!”.
  * 1a2. Use case ends.

**Use case 3: Delete Customer and All Orders Related to The Customer**

* **System**: SellSavvy

> * **Actor**: USer

Reviewer: What is USer?

* **Use Case**: UC03 - Delete Customer and All Orders related to the customer

**MSS**

1.  User wants to delete a customer.

> 2.  User chooses to view all customers.

Reviewer 2: Given link to `view all customers`, would linking to UC01 be better?

3.  SellSavvy displays the list of customers.
4.  User finds the customer index from the list.

> 5.  User deletes the customer by his index.

Reviewer 2: `his` may be slightly unclear. Perhaps specifying as `customer index` instead may be clearer?
```suggestion
5.  User deletes the customer by the customer's index in the list.
```

List here would follow from UC01's output list

6.  SellSavvy updates the list of customers and indicates that delete is successful.
7.  Use case ends.

**Use case 4: List Orders by Customer**

* **System**: SellSavvy

> * **Actor**: USer

Reviewer: What is USer?

* **Use Case**: UC04 - List Orders by Customer

**MSS**

1.  User wants to view all orders made by a specific customer
2.  User looks at the existing list for the customer name.

> 3.  User inputs command to list all orders, by the index of customer in contact list.

Reviewer 2: This is good, I like that source of customer index is clear.

Author: Thank you.

4.  SellSavvy validates the input, and retrieves a list of all orders made by specified customer.
5.  SellSavvy displays the orders in a list in GUI.
6.  Use case ends.

**Extensions**

* 3a. SellSavvy detects required details missing.
    * 3a1. SellSavvy displays an error message “Missing details of customer!”.
    * 3a2. Use case ends.

* 3b. User can list orders made by a customer using their username.
    * 3b1. User inputs command to list all orders using the username of the intended customer.
    * 3b2. Use case resumes from step 4.

**Use case 5: Mark Order as Completed**

* **System**: SellSavvy

> * **Actor**: USer

Reviewer: What is USer?

* **Use Case**: UC05 - Mark Order as Completed

**MSS**

1.  User wants to mark an order by a customer as completed.
2.  User chooses to view all customers.
3.  SellSavvy displays the list of customers.
4.  User finds the customer index from the list.
5.  User chooses to view all orders from the customer with that customer index.
6.  Sellsavvy displays the list of orders under the customer with that customer index.
7.  User marks the order as completed by the customer index and the order index.
8.  SellSavvy updates the status of the order and indicates that the action is successful.
9.  Use case ends.

**Extensions**

* 1a.  Remove marking of order
    * 1a1. User wants to remove marking on an order due to mistake.
    * 1a2. User chooses to view all customers.
    * 1a3. SellSavvy displays the list of customers.
    * 1a4. User finds the customer index from the list.
    * 1a5. User chooses to view all orders from the customer with that customer index.
    * 1a6. Sellsavvy displays the list of orders under the customer with that customer index.
    * 1a7. User inputs command to remove marking on the order by the customer index and the order index.

>     * 1a8. Use case resumes to step 8.

Reviewer 2: Consider removing list indexing where `use case resumes...`

**Use case 6: Delete an order**

* **System**: SellSavvy

> * **Actor**: USer

Reviewer: What is USer?
